internal/ingress/controller/kong.go

```diff
@@ -408,108 +408,96 @@ func (n *NGINXController) syncServices(ingressCfg *ingress.Configuration) (bool,
+					if !pluginDeepEqual(pluginInk8s.Config, &pluginInKong) ||
+						(!pluginInk8s.Disabled != pluginInKong.Enabled) ||
+						(pluginInk8s.ConsumerRef != "" && pluginInKong.Consumer == "") {
+						glog.Infof("plugin %v configuration in kong is outdated. Updating...", pluginInk8s.Name)
+						p := kongadminv1.Plugin{
+							Name:    pluginInKong.Name,
+							Config:  kongadminv1.Configuration(pluginInk8s.Config),
+							Enabled: pluginInk8s.Disabled,
 							Service: svc.ID,
-							Config:  kongadminv1.Configuration(k8sPlugin.Config),
 							Required: kongadminv1.Required{
-								ID: pluginID,
+								ID: pluginInKong.ID,
 							},
 						}
-
-						if k8sPlugin.ConsumerRef != "" {
-							consumer, err := n.store.GetKongConsumer(k8sPlugin.Namespace, k8sPlugin.ConsumerRef)
-							if err != nil {
-								glog.Errorf("Unexpected error searching plugin configuration %v for service %v: %v", plugin, svc.ID, err)
-							} else {
-								p.Consumer = fmt.Sprintf("%v", consumer.GetUID())
-							}
-						}
-
-						_, res := client.Plugins().CreateInService(svc.ID, p)
-						if res.StatusCode != http.StatusCreated {
-							return false, errors.Wrap(res.Error(), fmt.Sprintf("creating a Kong plugin %v in service %v", plugin, svc.ID))
-						}
-
-						continue
-					}
-
-					// check the kong plugin configuration is up to date
-					if !pluginDeepEqual(k8sPlugin.Config, configuredPlugin) ||
-						(!k8sPlugin.Disabled != configuredPlugin.Enabled) {
-						glog.Infof("plugin %v configuration in kong is outdated. Updating...", k8sPlugin.PluginName)
-						p := &kongadminv1.Plugin{
-							Name:    configuredPlugin.Name,
-							Config:  kongadminv1.Configuration(k8sPlugin.Config),
-							Enabled: k8sPlugin.Disabled,
-							Service: svc.ID,
-							Route:   "",
-						}
-
-						if k8sPlugin.ConsumerRef != "" && configuredPlugin.Consumer == "" {
-							consumer, err := n.store.GetKongConsumer(k8sPlugin.Namespace, k8sPlugin.ConsumerRef)
+						if pluginInk8s.ConsumerRef != "" {
+							consumer, err := n.store.GetKongConsumer(pluginInk8s.Namespace, pluginInk8s.ConsumerRef)
 							if err != nil {
-								glog.Errorf("Unexpected error searching plugin configuration %v for service %v: %v",
-									plugin, svc.ID, err)
+								glog.Errorf("Unexpected error searching for consumer %v: %v",
+									pluginInk8s.ConsumerRef, err)
 								continue
 							}
 							p.Consumer = fmt.Sprintf("%v", consumer.GetUID())
 						}
-						_, res := client.Plugins().Patch(configuredPlugin.ID, p)
-						if res.StatusCode != http.StatusOK {
-							glog.Errorf("Unexpected error updating plugin configuration %v for service %v: %v",
-								plugin, svc.ID, res)
-						}
-
-						continue
 					}
+				}
+			}
 
-					glog.Infof("plugin %v configuration in kong is up to date.", k8sPlugin.PluginName)
+			for _, plugin := range pluginsToDelete {
+				err := client.Plugins().Delete(plugin.ID)
```

Comment:
can we use the same logic as create and update that use statuscode to determine if delete is success or not?

Reply:
The API doesn't return an error right now.
We will be moving away from the bundled Kong Admin API so this will go away at some point but for now, we will have to stick with this interface.